RestaurantReservation.Db/Repositories/OrderRepository.cs

```diff
@@ -0,0 +1,47 @@
+﻿using Microsoft.EntityFrameworkCore;
+using RestaurantReservation.Db.Entities;
+using RestaurantReservation.Db.Repositories.IRepositories;
+
+namespace RestaurantReservation.Db.Repositories
+{
+    public class OrderRepository : EntityRepository<Order>, IOrderRepository
+    {
+
+        public OrderRepository(RestaurantReservationDbContext dbContext) : base(dbContext)
+        {
+        }
+
+        public async Task<decimal> CalculateAverageOrderAmountAsync(int employeeId)
+        {
+            var OrderAmountList = await _dbContext.Orders.AsNoTracking()
+                       .Where(o => o.EmployeeId == employeeId)
+                       .Include(o => o.OrderItems)
+                       .ThenInclude(oi=> oi.MenuItem)
+                       .AsSplitQuery()
+                       .ToListAsync();
```

Comment:
Please post the generated SQL query in the PR.

Reply:
info: 1/8/2024 09:10:13.025 RelationalEventId.CommandExecuted[20101] (Microsoft.EntityFrameworkCore.Database.Command)
      Executed DbCommand (103ms) [Parameters=[@__employeeId_0='4'], CommandType='Text', CommandTimeout='30']
      SELECT [o].[OrderId], [o].[EmployeeId], [o].[OrderDate], [o].[ReservationId]
      FROM [Orders] AS [o]
      WHERE [o].[EmployeeId] = @__employeeId_0
      ORDER BY [o].[OrderId]
info: 1/8/2024 09:10:31.579 RelationalEventId.CommandExecuted[20101] (Microsoft.EntityFrameworkCore.Database.Command)
      Executed DbCommand (68ms) [Parameters=[@__employeeId_0='4'], CommandType='Text', CommandTimeout='30']
      SELECT [t].[OrderItemId], [t].[MenuItemId], [t].[OrderId], [t].[Quantity], [t].[MenuItemId0], [t].[Description], [t].[Name], [t].[Price], [t].[RestaurantId], [o].[OrderId]
      FROM [Orders] AS [o]
      INNER JOIN (
          SELECT [o0].[OrderItemId], [o0].[MenuItemId], [o0].[OrderId], [o0].[Quantity], [m].[MenuItemId] AS [MenuItemId0], [m].[Description], [m].[Name], [m].[Price], [m].[RestaurantId]
          FROM [OrderItems] AS [o0]
          INNER JOIN [MenuItems] AS [m] ON [o0].[MenuItemId] = [m].[MenuItemId]
      ) AS [t] ON [o].[OrderId] = [t].[OrderId]
      WHERE [o].[EmployeeId] = @__employeeId_0
      ORDER BY [o].[OrderId]